check_limits.py

```diff
@@ -1,18 +1,36 @@
 
 def battery_is_ok(temperature, soc, charge_rate):
-  if temperature < 0 or temperature > 45:
-    print('Temperature is out of range!')
-    return False
-  elif soc < 20 or soc > 80:
-    print('State of Charge is out of range!')
-    return False
-  elif charge_rate > 0.8:
-    print('Charge rate is out of range!')
-    return False
+    return (temperature_is_ok(temperature) and soc_is_ok(soc) and charge_rate_is_ok(charge_rate))
 
-  return True
+
+def temperature_is_ok(temperature):
+    if temperature < 0 or temperature > 45:
+        print_text('Temperature is out of range!')
+        return False
+    else:
+        return True
+
+
+def soc_is_ok(soc):
+    if soc < 20 or soc > 80:
```

Comment:
Avoid repeated < `or` > checks by having them in a function e.g., `isWithinRange`. Pass the range to it as well.
This way only that function needs to be tested and re-used everywhere.